service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/create/GetAzureNetworkStep.java

```diff
@@ -0,0 +1,72 @@
+package bio.terra.workspace.service.resource.controlled.flight.create;
+
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.StepStatus;
+import bio.terra.stairway.exception.RetryException;
+import bio.terra.workspace.app.configuration.external.AzureConfiguration;
+import bio.terra.workspace.service.crl.CrlService;
+import bio.terra.workspace.service.resource.controlled.ControlledAzureNetworkResource;
+import bio.terra.workspace.service.resource.exception.DuplicateResourceException;
+import bio.terra.workspace.service.workspace.model.AzureCloudContext;
+import com.azure.core.management.exception.ManagementException;
+import com.azure.resourcemanager.compute.ComputeManager;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Gets an Azure Network, and fails if it already exists. This step is designed to run immediately
+ * before {@link CreateAzureNetworkStep} to ensure idempotency of the create operation.
+ */
+public class GetAzureNetworkStep implements Step {
+
+  private static final Logger logger = LoggerFactory.getLogger(GetAzureNetworkStep.class);
+  private final AzureConfiguration azureConfig;
+  private final CrlService crlService;
+  private final ControlledAzureNetworkResource resource;
+  private final AzureCloudContext azureCloudContext;
+
+  public GetAzureNetworkStep(
+      AzureConfiguration azureConfig,
+      AzureCloudContext azureCloudContext,
+      CrlService crlService,
+      ControlledAzureNetworkResource resource) {
+    this.azureConfig = azureConfig;
+    this.azureCloudContext = azureCloudContext;
+    this.crlService = crlService;
+    this.resource = resource;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
+    ComputeManager computeManager = crlService.getComputeManager(azureCloudContext, azureConfig);
+    try {
+      computeManager
+          .networkManager()
+          .networks()
+          .getByResourceGroup(
+              azureCloudContext.getAzureResourceGroupId(), resource.getNetworkName());
+      return new StepResult(
+          StepStatus.STEP_RESULT_FAILURE_FATAL,
+          new DuplicateResourceException(
+              String.format(
+                  "An Azure Network with name %s already exists in resource group %s",
+                  azureCloudContext.getAzureResourceGroupId(), resource.getNetworkName())));
+    } catch (ManagementException e) {
+      // Azure error codes can be found here:
+      // https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/common-deployment-errors
+      if (StringUtils.equals(e.getValue().getCode(), "ResourceNotFound")) {
+        return StepResult.getStepResultSuccess();
+      }
+      return new StepResult(StepStatus.STEP_RESULT_FAILURE_RETRY, e);
```

Comment:
Same question here - any better way to filter this than "all errors except... get retried"?
It is OK with me if the answer is: this is fine.

Reply:
answered in `SubnetsNotInSameVnet` comment above